plugins/module_utils/network/sonic/config/ospf_area/ospf_area.py

```diff
@@ -0,0 +1,1153 @@
+        requests.extend(stub_delete_requests)
+
+        # This section, for 'propagation endpoint', handles the inter-area propagation policies
+        # this doesn't use a "propagation_all_gone" flag because the nested subection ranges are dealt with separately above.
+        # The other settings of filter lists are nested in propagation in REST API but are in root area settings for argspec.
+        # This means that a simple check that commands and want (assuming this was passed commands that are in and the same value as have)
+        # are the same length is enough to cover the same behaviors
+        propagation_delete_requests, propagation_deleted_area = self.build_area_delete_propagation_requests(
+            commands,
+            have,
+            ranges_all_gone,
+            ranges_delete_requests)
+        requests.extend(propagation_delete_requests)
+        area_already_deleted = area_already_deleted or propagation_deleted_area
+
+        if delete_everything or \
+                (len(commands) == len(have) and stub_all_gone and vlink_all_gone and ranges_all_gone and networks_all_gone):
+            # delete all settings for area.
+            # either only area id was specified or any form of clear data,
+            # or all settings are named and for the more complex nested subsections of argspec,
+            # those subsections also match and are cleared (need the extra flag since length only compares the subsections existence not contents)
+            # there are cases where deleting sub-sections causes area to disappear. need to make sure to delete
+            # area too so need to check when adding that is needed
+            if not area_already_deleted:
+                requests.append({'path': self.ospf_area_uri.format(vrf=commands["vrf_name"], area_id=commands["area_id"]), 'method': 'DELETE'})
+            return requests
+        return requests
+
+    def build_area_stub_delete_requests(self, request_root, commands, have):
+        '''builds the requests to delete a single area's stub config. takes a pair of single area commands and have.
+        returns a tuple of whether everything in stub config was deleted and requests to cause changes'''
+        requests = []
+        relevant_have = {"default_cost": have.get("default_cost", None),
+                         "no_summary": have.get('stub', {}).get("no_summary", None),
+                         "enabled": have.get('stub', {}).get("enabled", None)}
+        relevant_commands = {"default_cost": commands.get("default_cost", None),
+                             "no_summary": commands.get('stub', {}).get("no_summary", None),
+                             "enabled": commands.get('stub', {}).get("enabled", None)}
+        relevant_have = remove_empties(relevant_have)
+        relevant_commands = remove_empties(relevant_commands)
+
+        if "default_cost" in relevant_commands:
```

Comment:
Is there a reason why this and the following two 'if \<x\> in relevant_commands' checks aren't also checking for the "anded" condition of '\<x\> in relevant_have"? This is usually done to avoid unnecessary sending of REST APIs to delete configuration that doesn't exist. 

e.g.

```suggestion
        if "default_cost" in relevant_commands and "default_cost" in relevant_have:
```

Reply:
could check but the cases where the method is called already limit commands to what is already in have. Deleted state commands is only what is the same between want and have, replaced and overridden if there's a reason to delete an area, it uses the current have as the delete commands. merged doesn't call delete.

Reply:
Thanks for pointing this out. Your response answers my question. I agree that the additional check is not needed. 

Resolved.